### ISSUE TITLE
[UWP] Image render throws unhandled exception

### ISSUE THREAD

**Issue by nimercer** [microsoft/AdaptiveCards#3026]
# Platform

What platform is your issue or question related to? (Delete other platforms).
* UWP

# Author or host
Host

If you're an author, who are you sending cards to?

# Version of SDK
1.2.0-beta1

# Issue

An exception is thrown inside an async operation inside SetImageOnUIElement function in the XamlBuilder. (While rendering an adaptive image). As the call gets dispatched to another thread, it doesn't get caught by the usual mechanism that means exceptions don't throw inside Render calls and causes a crash inside the host app.

See mail thread: Crash on Adaptive Card Image Renderer for Watson

**Comment by shalinijoshi19** [comment]
@paulcam206 / @RebeccaAnne/ @almedina-ms  let's just use this issue to track the overall async operation improvements in XamlBuilder including related issues #491 , #991, #2396 and the following improvement suggested by the Timeline folks: 

<snip>
TL;DR: We may want to consider refactoring XamlBuilder::PopulateImageFromUrlAsync to use tasks that we can chain with .then instead of nested put_Complete callbacks for finer-grained control of the exception handling.
</snip>

**Comment by shalinijoshi19** [comment]
Tracking this issue for all XamlBuilder async work processing/exception handling improvements including switching to an async task based model here:

<snip>
TL;DR: We may want to consider refactoring XamlBuilder::PopulateImageFromUrlAsync to use tasks that we can chain with .then instead of nested put_Complete callbacks for finer-grained control of the exception handling.
</snip>

**Comment by shalinijoshi19** [comment]
Related #2396 , #491

**Comment by shalinijoshi19** [comment]
@paulcam206  load-balancing to @almedin-ms for tackling this so he is unblocked on #2712

**Comment by paulcam206** [comment]
Also related: #2534

**Comment by shalinijoshi19** [comment]
@paulcam206 assigning to you; Do you just want to track #2534 as part of fixingn all these issues instead of multiple ones? And these could then be tests for hte rework you are doing as part of 2534?

**Comment by paulcam206** [comment]
This is possible to address now that we have cppwinrt support in the UWP codebase (e.g. now we can utilize `.then`)

**Comment by nimercer** [comment]
@paulcam206 Great! For clarification, is this something that you're going to address in the next release? Do you have an eta? :)

**Comment by jonmill** [comment]
This should be fixed in the newest build; please re-open if this is not the case and we'll engage. Thanks!
